src/container/media.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+import { Breakpoint, BREAKPOINTS_DESCENDING, matchBreakpointMapping } from '../internal/breakpoints';
+import { ContainerProps, MediaDefinition } from './interfaces';
+import { useContainerBreakpoints } from '../internal/hooks/container-queries';
+
+type MediaProps = keyof ContainerProps.Media;
+
+const MEDIA_PROPS: MediaProps[] = ['position', 'width', 'height', 'content'];
+
+const hasDefinedBreakpoint = (obj: any) => {
+  return BREAKPOINTS_DESCENDING.some(key => obj[key] !== undefined);
+};
+
+export function useMedia(media?: ContainerProps.Media) {
+  const initialState = {
+    content: media?.content,
+    height: '',
+    width: '',
+    position: media?.position || 'top',
+  };
+  const [mediaState, setMediaState] = useState(initialState);
+  const [breakpoint, breakpointRef] = useContainerBreakpoints(getBreakpointsForMedia(media));
+  useEffect(() => {
+    if (!media || !breakpoint) {
+      return;
+    }
+
+    setMediaState(prevMediaState => {
```

Comment:
Please consider if you really need a `useEffect` hook here: https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes.

Reply:
Indeed! Great call - no need for state even.